src/honeysql_postgres/format.cljc

```diff
@@ -143,11 +145,40 @@
            util/get-first
            sqlf/to-sql)))
 
-(defmethod format-clause :with-columns [[_ columns] _]
-  (sqlf/paren-wrap (->> columns
-                        util/get-first
-                        (map #(sqlf/space-join (map sqlf/to-sql %)))
-                        sqlf/comma-join)))
+(def ^:private constraints-format-map
+  {:primary-key "PRIMARY KEY"
+   :unique      "UNIQUE"})
+
+(defn- format-constraint-clause
+  [constraint]
+  (let [[constraint-type constraint-args] constraint]
```

Comment:
It'd be nicer to move the sequential destructuring to params.

Reply:
Done!